### PR TITLE
Add support for ubuntu jammy (22.04)

### PR DESCRIPTION
**Pull request by matthewoliver**
This patch adds jammy support to VSAIO and also makes it the default.

Also default to use_python3

### CODE REVIEW

**Comment by clayg** [comment]
@tipabu do you remeber what issue you had when you tested jammy support back closer to the initial release?

**Comment by matthewoliver** [comment]
Added a second patch that adds the py37 - py39 deadsnake pythons. Debugged them, and now also install their distutils packages as well, which fixes the pip install issues using them.

**Review by clayg — COMMENTED** [review]
sorry, these were left from a couple of days ago and I don't know how to github

File: Vagrantfile [hunk 1]
```diff
@@ -7,10 +7,11 @@
 
 require 'ipaddr'
 
-DEFAULT_BOX = "bionic"
+DEFAULT_BOX = "jammy"
 
 # Note: 18.04/bionic requires Vagrant 2.02 or newer because 18.04 ships without ifup/ifdown by default.
 vagrant_boxes = {
+  "jammy" => "http://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64-vagrant.box",
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply Vagrantfile:14]: oh yeah. I'm not an Ubuntu guy so no idea the order of naming. But should've guessed/noticed :)

File: Vagrantfile [hunk 2]
```diff
@@ -124,10 +125,12 @@ Vagrant.configure("2") do |global_config|
       end
 
       # Install libssl for Chef (https://github.com/hashicorp/vagrant/issues/10914)
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply Vagrantfile:127]: Done

File: cookbooks/swift/recipes/setup.rb [hunk 3]
```diff
@@ -55,15 +55,25 @@
...
-  "libssl-dev", # libssl-dev is required for building wheels from the cryptography package in swift.
-  "curl", "gcc", "memcached", "rsync", "sqlite3", "xfsprogs", "git-core", "build-essential",
-  "python-dev", "libffi-dev", "python3.5", "python3.5-dev",
-  "python3.6", "python3.6-dev", "python3.7", "python3.7-dev",
- "python3.8", "python3.8-dev",
-  "libxml2-dev", "libxml2", "libxslt1-dev", "zlib1g-dev", "autoconf", "libtool",
-  "openjdk-11-jre-headless", "haproxy",
-]
+if node['platform_version'] == '22.04'
+  required_packages = [
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply cookbooks/swift/recipes/setup.rb:59]: oh yeah, that makes more sense.

File: localrc-template [hunk 4]
```diff
@@ -25,17 +25,17 @@ export SERVERS_PER_PORT=0  # 1 port per dev in ring; N obj-servers per port
...
-export SWIFTCLIENT_REPO=git://github.com/openstack/python-swiftclient.git
-export SWIFTBENCH_REPO=git://github.com/openstack/swift-bench.git
-export LIBERASURECODE_REPO=git://github.com/openstack/liberasurecode.git
-export PYECLIB_REPO=git://github.com/openstack/pyeclib.git
-export SWIFTZIPKIN_REPO=git://github.com/swiftstack/swift-zipkin.git
+export SWIFTCLIENT_REPO=https://github.com/openstack/python-swiftclient
+export SWIFTBENCH_REPO=https://github.com/openstack/swift-bench
+export LIBERASURECODE_REPO=https://github.com/openstack/liberasurecode
+export PYECLIB_REPO=https://github.com/openstack/pyeclib
+export SWIFTZIPKIN_REPO=https://github.com/swiftstack/swift-zipkin
```
[the remark this answers is not in the document]
Comment (matthewoliver) [reply localrc-template:38]: Me too, but Github was erroring out when I tried to access it via git addresses, they don't seem to allow that access anymore. But https worked.

[the remark this answers is not in the document]
Comment (clayg) [reply localrc-template:38]: the git://github.com/org/project.git protocol has been deprecated, but git@github.com:org/project.git still works

the key forwarding should work by default, but guess using https ensures people w/o a github account can still use the defautl saio options (and we only push to gerrit anyway, so I guess it's fine)

**Comment by clayg** [comment]
This didn't work for me w/o

```
vagrant@saio:/vagrant/python-swiftclient$ cat /etc/gitconfig
[safe]
	directory = *
```

Something to do with a recent git change, devstack had the same problem:

https://review.opendev.org/c/openstack/devstack/+/837749/3/functions-common#601

**Comment by matthewoliver** [comment]
Shall I throw that into the chef recipe then! I just came across it too.

**Comment by clayg** [comment]
Some kind of solution will be needed to "support" ubuntu jammy - the safe.directory thing seems to work.

Since, it's inside the vm; I think it's fine from a safety persepctive - I think the problem shows up because of something to do with the user during provisioning: root vs. vagrant.

If a better solution comes along we always update our cookbooks.

**Comment by matthewoliver** [comment]
Oh I make the cookbook change last night, must have forgotten to push it.
Will do that when I get in front of my computer next.

**Comment by matthewoliver** [comment]
Added a commit that adds the /etc/gitconfig to the cookbook.

**Comment by clayg** [comment]
the pbr install still failed for me.  it looks like we install from source before the configs playbook runs [1]

I pushed up a change that works for me

https://github.com/NVIDIA/vagrant-swift-all-in-one/commit/34d4f6de8deea585b08150ab73abbaa314b153af

I think it'd be easier to collaborate if we use branches on the origin instead of our forks.

1. https://github.com/NVIDIA/vagrant-swift-all-in-one/blob/master/cookbooks/swift/recipes/default.rb

**Comment by matthewoliver** [comment]
oh yeah totally, I thought I tried that but I don't think I had rights too.

**Comment by matthewoliver** [comment]
Happy for this to be moved into a branch and them merged in. Feel free to close this if we want to work on it (or land it) from the jammy_works branch.

**Comment by clayg** [comment]
Ok, great - I added you as an project admin and created https://github.com/NVIDIA/vagrant-swift-all-in-one/pull/127

**Pull request closed without merging**